pq-crypto/kyber_r3/kyber512r3_cbd.c

```diff
@@ -3,7 +3,7 @@
 #include "kyber512r3_cbd.h"
 
 #if S2N_ANY_NONPORTABLE_OPTIMIZATIONS_ENABLED
-#error "Compiling portable code with non-portable assembly optimizations is not allowed"
+#error "Compiling portable code with non-portable assembly optimizations is dangerous. If you know that every CPU using these build artifacts actually has these non-portable instructions, then you can enable S2N_ALLOW_NONPORTABLE_OPTIMIZATIONS flag"
```

Comment:
Why do we want s2n policing `CFLAGS`?  `S2N_ALLOW_NONPORTABLE_OPTIMIZATIONS` is a work-around, but imagine every single library having similar logic, then compiling with hardware optimizations will be nearly impossible without figuring out tweaks for each of those libraries. None of other commonly used libraries try to double-check if non-portable optimizations are desired, why s2n is special? Also consider that on a lot of linux distributions having `CFLAGS="-march=native"` is common for locally built packages, as they are never distributed and only used withing the same hardware. This change will create additional obstacles for building s2n on such platforms.

Reply:
I agree here... This problem isn't at all specific to s2n. It goes beyond the PQ stuff as well. Even basic loops anywhere can be vectorized by the compiler under the right conditions. I think if the application has set the cflags to a specific architecture they shouldn't have to jump through additional hoops to get those optimizations.

Reply:
The intent was for these checks to be a mechanism to catch any potential future bugs in s2n's build time script that inject the AVX flags, rather than relying on good intentions of reviewers catching these issues during PR. This issue has been missed previously during review in a previous PR, so having a mechanism to check for it seems preferable.

Any thoughts on a different mechanism to catch issues where s2n's build scripts are incorrectly deciding to add AVX flags to portable code that shouldn't have them? Or do we want to remove all mechanisms to catch this issue in favor of good intentions and code review?

Reply:
I think we can make these checks opt-in rather than opt-out. We would then enable them in CI to make sure we're not doing anything ourselves to enable them.